### ISSUE TITLE
Training code or equation for loss function

### ISSUE THREAD

**Issue by KohakuBlueleaf** [madebyollin/taesd#2]
Is it possible to opensource the training code or the loss equation?
I'm considering to train different scale TAESD (like 1M 2.5M(this) 10M) to check if we can get a series of models that fit for different usage.

**Comment by madebyollin** [comment]
The TAESD encoder is just trained with MSE (against the SD encoder results), and the TAESD decoder is trained as a conditional GAN (conditioned on the SD encoder results).

Regarding how to train conditional GANs, you could look at the GAN training code used in SD https://github.com/CompVis/latent-diffusion#training-autoencoder-models

**Comment by KohakuBlueleaf** [comment]
As I know
the taming transformer training code has both LPIPS loss + mse loss + discriminator loss
do you also use this three?

And thx for this information!

**Comment by madebyollin** [comment]
I only used discriminator and tiny bit of MSE (didn't try LPIPS)

**Comment by KohakuBlueleaf** [comment]
> I only used discriminator and tiny bit of MSE (didn't try LPIPS)

Ok!
Thx!

**Comment by madebyollin** [comment]
As part of TAESD 1.1, I added LPIPS (though still at lower weight than adversarial). So it's overall not very different from the original VQGAN approach.

**Comment by Manojbhat09** [comment]
hi, what do you mean by tiny bit of MSE ? there is only MSE at the decoder loss if I am not wrong?

**Comment by KohakuBlueleaf** [comment]
> hi, what do you mean by tiny bit of MSE ? there is only MSE at the decoder loss if I am not wrong?

Real img → encoder → Latent → decoder → recon img

loss = a * mse(recon, real) + b * lpips(recon, real) + c * discrininator(recon, real)

And we use very small a here so we said "tiny bit of mse"
